skiko/src/awtMain/kotlin/org/jetbrains/skiko/redrawer/Direct3DRedrawer.kt

```diff
@@ -84,10 +84,14 @@ internal class Direct3DRedrawer(
     }
 
     private fun drawAndSwap(withVsync: Boolean) = synchronized(drawLock) {
-        if (!isDisposed) {
-            contextHandler.draw()
-            swap(device, withVsync)
+        if (isDisposed) {
+            return
         }
+        if (layer.width <= 0 || layer.height <= 0) {
```

Comment:
`layer.width` is AWT-manipulated property, and we can have a race condition when accessing it from another thread (`dispatcherToBlockOn`).

Better to move it outside of `withContext(dispatcherToBlockOn) {`. Or remember the local size inside `resizeBuffers`/`makeDirectXSurface` (I vote for it)